src/formats/internal/memreader.c

```diff
@@ -62,14 +62,22 @@ uint8_t memread_ubyte(memreader *reader) {
 
 uint16_t memread_uword(memreader *reader) {
     uint16_t r;
+#ifdef BIG_ENDIAN_BUILD
+    r = __builtin_bswap16(*((uint16_t *)(reader->buf + reader->pos)));
+#else
     memcpy(&r, reader->buf + reader->pos, sizeof(r));
+#endif
     reader->pos += sizeof(r);
     return r;
```

Comment:
buf+pos is not insufficiently aligned to be a (uint16_t *)

please use the memcpy from the existing code, and keep your #ifdef block as small as possible (your sd_peek_dword is a good example of how it should look).

Reply:
I've thought about this, and it hasn't been an issue with the current game assets. I don't want to slow it down more than needed.